internals/overlord/cmdstate/handlers.go

```diff
@@ -252,7 +252,12 @@ func (e *execution) do(ctx context.Context, task *state.Task) error {
 			// websocket and write to the PTY.
 			go func() {
 				<-wsutil.WebsocketRecvStream(master, ioConn)
-				master.Close()
+				// If the interactive is enforced, it is possible to finish
+				// reading earlier than the mirroring go routine sends all the
+				// output to the client. Thus, closing the master descriptor
+				// here will terminate mirroring prematurely. Instead, we
+				// should send Ctrl-D to the fd to indicate the end of input.
+				master.Write([]byte{byte(unix.VEOF)})
```

Comment:
The fix makes sense, see e.g. https://stackoverflow.com/a/18366397/705086

I do wonder about some corner cases, e.g.:

- `pebble -i yada yada < somefile` where input is not from a terminal, but the output is
- all of stdin/stdout/stderr are redirected, and yet `-i` is supplied

Please consider how to test these -- unit tests? functional? one-off manual tests? I'm not sure what would be more appropriate.

Reply:
Yeah, it's not ideal, but we currently don't have any automated tests for interactive mode. We have on our roadmap for @IronCore864 to write end-to-end tests for `pebble run`; maybe some `pebble exec -i` tests can be included in that work?

In any case, I've done some pretty extensive manual testing of all the various exec modes, and this fix seems to work nicely. I've also tested that the number of file descriptors isn't increasing (leaking) by watching `/bin/sh -c 'ls /proc/{pebble_pid}/fd/ | wc -l'`, and it's stable (stays at 10).

Tiexin has also done manual testing on this PR.